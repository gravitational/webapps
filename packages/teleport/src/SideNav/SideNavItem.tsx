/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import styled from 'styled-components';
import defaultTheme from 'design/theme';
import { color } from 'design/system';

const fromTheme = ({ $nested = false, theme = defaultTheme }) => {
  const css = {
    fontSize: '14px',
    fontWeight: theme.regular,
    fontFamily: theme.font,
    paddingLeft: theme.space[9] + 'px',
    paddingRight: theme.space[5] + 'px',
    background: theme.colors.primary.light,
    color: theme.colors.text.secondary,
    minHeight: '56px',
    '&:active, &.active': {
      borderLeftColor: theme.colors.accent,
      background: theme.colors.primary.lighter,
      color: theme.colors.primary.contrastText,
    },

    '&:hover, &:focus': {
      background: theme.colors.primary.lighter,
      color: theme.colors.primary.contrastText,
    },
  };

  if ($nested) {
    css.fontWeight = theme.regular;
    css.paddingLeft = theme.space[10] + 'px';
    css.minHeight = '48px';
  }

  return css;
};

const SideNavItem = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border: none;
  border-left: 4px solid transparent;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
  line-height: 24px;
  ${fromTheme}
  ${color}
`;

export default SideNavItem;
