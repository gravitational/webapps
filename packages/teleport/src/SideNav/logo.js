import logoSvg from 'design/assets/images/teleport-logo.svg';
export default logoSvg;
