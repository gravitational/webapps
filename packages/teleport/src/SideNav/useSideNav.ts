/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import { useMemo } from 'react';
import { useHistory } from 'react-router';
import useTeleport from 'teleport/useTeleport';
import useStickyClusterId from 'teleport/useStickyClusterId';
import * as Icons from 'design/Icon';
import * as Store from 'teleport/stores/storeNav';

export default function useSideNav() {
  const h = useHistory();
  const ctx = useTeleport();
  const { clusterId } = useStickyClusterId();
  const items = useMemo(
    () => makeItems(clusterId, ctx.storeNav.getSideItems()),
    [clusterId]
  );

  return {
    items,
    path: h.location.pathname,
  };
}

function makeItems(clusterId: string, storeItems: Store.NavItem[]) {
  const items = new Map<string, Item>();
  const itemGroups = getGroups();

  for (let i = 0; i < storeItems.length; i++) {
    const cur = storeItems[i];
    const groupName = cur.group;
    const item = {
      items: [] as Item[],
      route: cur.getLink(clusterId),
      exact: cur.exact,
      title: cur.title,
      Icon: cur.Icon,
    };

    if (itemGroups[groupName]) {
      itemGroups[groupName].items.push(item);
      items.set(groupName, itemGroups[groupName]);
    } else {
      items.set(i + '', item);
    }
  }

  return Array.from(items.values());
}

function getGroups() {
  return {
    team: {
      Icon: Icons.Users,
      title: 'Team',
      items: [] as Item[],
      route: '',
    },
    activity: {
      Icon: Icons.AlarmRing,
      title: 'Activity',
      items: [] as Item[],
      route: '',
    },
    clusters: {
      Icon: Icons.Clusters,
      title: 'Clusters',
      items: [] as Item[],
      route: '',
    },
  };
}

export interface Item {
  items: Item[];
  route: string;
  exact?: boolean;
  title: string;
  Icon: any;
}
