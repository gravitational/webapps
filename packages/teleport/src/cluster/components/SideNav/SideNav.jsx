/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import { NavLink } from 'react-router-dom';
import { withState } from 'shared/hooks';
import { SideNav, SideNavItem } from 'design';
import SideNavItemIcon from 'design/SideNav/SideNavItemIcon';
import { useStoreUser, useStoreNav } from 'teleport/teleportContextProvider';
import cfg from 'teleport/config';

export function ClusterSideNav({ items }) {
  const $items = items.map((item, index) => (
    <SideNavItem key={index} as={NavLink} exact={item.exact} to={item.to}>
      <SideNavItemIcon as={item.Icon} />
      {item.title}
    </SideNavItem>
  ));

  return <SideNav>{$items}</SideNav>;
}

export default withState(() => {
  const items = useStoreNav().getSideItems();
  const { version } = useStoreUser().state;

  return {
    version,
    items,
    clusterName: cfg.clusterName,
    homeUrl: cfg.getClusterRoute(),
  };
})(ClusterSideNav);
