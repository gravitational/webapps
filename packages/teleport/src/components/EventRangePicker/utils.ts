import moment from 'moment';

export function getRangeOptions(): EventRange[] {
  return [
    {
      name: 'Today',
      from: moment(new Date())
        .startOf('day')
        .toDate(),
      to: moment(new Date())
        .endOf('day')
        .toDate(),
    },
    {
      name: '7 days',
      from: moment()
        .subtract(6, 'day')
        .startOf('day')
        .toDate(),
      to: moment(new Date())
        .endOf('day')
        .toDate(),
    },
    {
      name: 'Custom Range...',
      isCustom: true,
      from: new Date(),
      to: new Date(),
    },
  ];
}

export type EventRange = {
  from: Date;
  to: Date;
  isCustom?: boolean;
  name?: string;
};
