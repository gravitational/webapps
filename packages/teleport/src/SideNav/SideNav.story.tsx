/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';
import * as Icons from 'design/Icon';
import { Box } from 'design';

import { SideNav } from './SideNav';
import { Item } from './useSideNav';

export default {
  title: 'Teleport/SideNav',
};

export const Story = () => {
  const props = {
    ...defaultProps,
  };

  const inMemoryHistory = createMemoryHistory({
    initialEntries: ['/web/roles'],
    initialIndex: 0,
  });
  return (
    <Box
      mt={-3}
      mx={-3}
      height="100%"
      style={{ position: 'absolute', overflow: 'hidden' }}
    >
      <Router history={inMemoryHistory}>
        <SideNav {...props} />
      </Router>
    </Box>
  );
};

const defaultProps = {
  path: '/web/roles',
  items: [
    {
      items: [],
      route: '/web/cluster/one/nodes',
      Icon: Icons.Server,
      exact: true,
      title: 'Servers',
    },
    {
      items: [],
      route: '/web/cluster/one/apps',
      Icon: Icons.NewTab,
      exact: true,
      title: 'Applications',
    },
    {
      items: [],
      route: '/web/cluster/one/kubes',
      Icon: Icons.Kubernetes,
      exact: true,
      title: 'Kubernetes',
    },
    {
      title: 'Team',
      Icon: Icons.Users,
      items: [
        {
          items: [],
          route: '/web/users',
          Icon: Icons.Users,
          exact: true,
          title: 'Users',
        },
        {
          items: [],
          route: '/web/roles',
          Icon: Icons.Profile,
          exact: true,
          title: 'Roles',
        },
        {
          items: [],
          route: '/web/sso',
          Icon: Icons.Lock,
          exact: false,
          title: 'Auth. Connectors',
        },
      ],
      route: '',
    },
    {
      title: 'Activity',
      Icon: Icons.AlarmRing,
      items: [
        {
          items: [],
          route: '/web/cluster/one/sessions',
          Icon: Icons.Terminal,
          exact: true,
          title: 'Active Sessions',
        },
        {
          items: [],
          route: '/web/cluster/one/recordings',
          Icon: Icons.CirclePlay,
          exact: true,
          title: 'Session Recordings',
        },
        {
          items: [],
          route: '/web/cluster/one/audit',
          Icon: Icons.ListThin,
          title: 'Audit Log',
        },
      ],
      route: '',
    },
    {
      title: 'Clusters',
      Icon: Icons.Clusters,

      items: [
        {
          items: [],
          route: '/web/clusters',
          Icon: Icons.EqualizerVertical,
          exact: false,
          title: 'Manage Clusters',
        },
        {
          items: [],
          route: '/web/trust',
          Icon: Icons.Cluster,
          title: 'Trust',
        },
      ],
      route: '',
    },
  ] as Item[],
};
