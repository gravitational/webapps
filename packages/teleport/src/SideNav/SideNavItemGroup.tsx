/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { matchPath } from 'react-router';

import * as Icons from 'design/Icon';

import SideNavItemIcon from './SideNavItemIcon';
import SideNavItem from './SideNavItem';
import { Item } from './useSideNav';

const SideNavItemGroup: React.FC<{ path: string; item: Item }> = props => {
  const { item, path } = props;
  const hasSelectedChild = isChildActive(path, item);
  // Auto expand on initial render if a child is selected.
  const [expanded, setExpanded] = React.useState(() =>
    isChildActive(path, item)
  );

  React.useEffect(() => {
    // Ensures parent is expanded, if a child is selected.
    if (hasSelectedChild && !expanded) {
      setExpanded(true);
    }
  }, [hasSelectedChild]);

  const ArrowIcon = expanded ? Icons.ArrowDown : Icons.ArrowRight;
  const style = {
    display: expanded ? 'block' : 'none',
  };

  const $children = item.items.map((i, index) => {
    return (
      <SideNavItem
        key={index}
        $nested={true}
        as={NavLink}
        exact={i.exact}
        to={i.route}
      >
        <StyledMarker className="marker"></StyledMarker>
        <SideNavItemIcon as={i.Icon} fontSize="2" mr={2} />
        {i.title}
      </SideNavItem>
    );
  });

  const className = hasSelectedChild ? 'actives' : '';

  return (
    <>
      <StyledGroup
        className={className}
        as="button"
        onClick={() => setExpanded(!expanded)}
      >
        <SideNavItemIcon as={item.Icon} />
        {item.title}
        <ArrowIcon
          ml="auto"
          mr={-2}
          color="inherit"
          style={{ fontSize: '14px' }}
        />
      </StyledGroup>
      <StyledChildrenContainer style={style}>
        {$children}
      </StyledChildrenContainer>
    </>
  );
};

export default SideNavItemGroup;

function isChildActive(url: string, item: Item) {
  return item.items.some(
    i =>
      !!matchPath(url, {
        path: i.route,
        exact: i.exact,
      })
  );
}

const fromTheme = ({ theme }) => {
  return {
    fontSize: '12px',
    fontWeight: theme.regular,
    fontFamily: theme.font,
    paddingLeft: theme.space[9] + 'px',
    paddingRight: theme.space[5] + 'px',
    background: theme.colors.primary.light,
    color: theme.colors.text.secondary,

    '&.active': {
      borderLeftColor: theme.colors.accent,
      background: theme.colors.primary.lighter,
      color: theme.colors.primary.contrastText,
      '.marker': {
        background: theme.colors.secondary.light,
      },
    },

    '&:hover': {
      background: theme.colors.primary.lighter,
    },

    '&:hover, &:focus': {
      color: theme.colors.primary.contrastText,
    },

    minHeight: '56px',
  };
};

const StyledChildrenContainer = styled.div`
  background: ${props =>
    `linefar-gradient(140deg, ${props.theme.colors.primary.lighter}, ${props.theme.colors.primary.light});`};
`;

const StyledMarker = styled.div`
  height: 8px;
  width: 8px;
  position: absolute;
  top: 16px;
  left: 26px;
`;

const StyledGroup = styled.div`
  margin: 0;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border: none;
  border-left: 4px solid transparent;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
  line-height: 24px;
  ${fromTheme}
`;
