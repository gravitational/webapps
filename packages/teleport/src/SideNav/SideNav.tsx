/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import styled from 'styled-components';
import { NavLink, Link } from 'react-router-dom';
import { Flex, Image } from 'design';

import cfg from 'teleport/config';

import SideNavItemIcon from './SideNavItemIcon';
import SideNavItem from './SideNavItem';
import SideNavItemGroup from './SideNavItemGroup';
import logoSvg from './logo';
import useSideNav from './useSideNav';

export default function Container() {
  const state = useSideNav();
  return <SideNav {...state} />;
}

export function SideNav(props: ReturnType<typeof useSideNav>) {
  const { items, path } = props;
  const $items = items.map((item, index) => {
    const isChild = item.items.length > 0;
    if (isChild) {
      return <SideNavItemGroup path={path} item={item} key={index} />;
    }

    return (
      <SideNavItem key={index} as={NavLink} exact={item.exact} to={item.route}>
        <SideNavItemIcon as={item.Icon} />
        {item.title}
      </SideNavItem>
    );
  });

  return (
    <Nav>
      <Logo />
      <Content>{$items}</Content>
    </Nav>
  );
}

export const Logo = () => (
  <LogoItem pl="4" width="208px" as={Link} to={cfg.routes.root}>
    <Image src={logoSvg} mx="3" maxHeight="24px" maxWidth="160px" />
  </LogoItem>
);

const LogoItem = styled(Flex)(
  props => `
  min-height: 56px;
  align-items: center;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
  &:hover {
    background ${props.theme.colors.primary.lighter};
    color ${props.theme.colors.primary.contrastText};
  }
`
);

export const Nav = styled.nav`
  background: ${props => props.theme.colors.primary.light};
  border-right: 1px solid ${props => props.theme.colors.primary.dark};
  min-width: 240px;
  width: 240px;
  overflow: auto;
  height: 100%;
  display: flex;
  flex-direction: column;
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  overflow: auto;
`;
