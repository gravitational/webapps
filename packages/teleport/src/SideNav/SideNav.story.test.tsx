import React from 'react';
import { Story } from './SideNav.story';
import { render } from 'design/utils/testing';

test('rendering of SideNav', () => {
  const { container } = render(<Story />);
  expect(container.firstChild).toMatchSnapshot();
});
