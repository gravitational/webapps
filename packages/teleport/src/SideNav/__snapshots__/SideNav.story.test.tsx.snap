// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of SideNav 1`] = `
.c6 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: -40px;
  margin-right: 16px;
  color: inherit;
  font-size: 16px;
}

.c9 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: auto;
  margin-right: -8px;
  color: inherit;
}

.c13 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: -40px;
  margin-right: 8px;
  color: inherit;
  font-size: 14px;
}

.c0 {
  box-sizing: border-box;
  margin-top: -16px;
  margin-left: -16px;
  margin-right: -16px;
  height: 100%;
}

.c3 {
  display: block;
  outline: none;
  margin-left: 16px;
  margin-right: 16px;
  max-width: 160px;
  max-height: 24px;
}

.c7 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: -40px;
  margin-right: 16px;
  color: inherit;
  font-size: 16px;
}

.c14 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: -40px;
  margin-right: 8px;
  color: inherit;
  font-size: 14px;
}

.c5 {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border: none;
  border-left: 4px solid transparent;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
  line-height: 24px;
  position: relative;
  font-size: 12px;
  font-weight: 400;
  font-family: Ubuntu2,-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  padding-left: 64px;
  padding-right: 32px;
  background: #222C59;
  color: rgba(255,255,255,0.56);
  min-height: 56px;
}

.c5:active,
.c5.active {
  border-left-color: #651FFF;
  background: #2C3A73;
  color: #FFFFFF;
  font-weight: 600;
}

.c5:active .marker,
.c5.active .marker {
  background: #651FFF;
}

.c5:hover {
  background: #2C3A73;
}

.c5:focus,
.c5:hover {
  color: #FFFFFF;
}

.c11 {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border: none;
  border-left: 4px solid transparent;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
  line-height: 24px;
  position: relative;
  font-size: 11px;
  font-weight: 400;
  font-family: Ubuntu2,-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  padding-left: 96px;
  padding-right: 32px;
  background: none;
  color: rgba(255,255,255,0.56);
  min-height: 40px;
}

.c11:active,
.c11.active {
  border-left-color: none;
  background: #2C3A73;
  color: #FFFFFF;
  font-weight: 600;
}

.c11:active .marker,
.c11.active .marker {
  background: #651FFF;
}

.c11:hover {
  background: #2C3A73;
}

.c11:focus,
.c11:hover {
  color: #FFFFFF;
}

.c10 {
  background: linefar-gradient(140deg,#2C3A73,#222C59);
}

.c12 {
  height: 8px;
  width: 8px;
  position: absolute;
  top: 16px;
  left: 26px;
}

.c8 {
  margin: 0;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border: none;
  border-left: 4px solid transparent;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
  line-height: 24px;
  font-size: 12px;
  font-weight: 400;
  font-family: Ubuntu2,-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  padding-left: 64px;
  padding-right: 32px;
  background: #222C59;
  color: rgba(255,255,255,0.56);
  min-height: 56px;
}

.c8.active {
  border-left-color: #651FFF;
  background: #2C3A73;
  color: #FFFFFF;
}

.c8.active .marker {
  background: #651FFF;
}

.c8:hover {
  background: #2C3A73;
}

.c8:hover,
.c8:focus {
  color: #FFFFFF;
}

.c2 {
  box-sizing: border-box;
  padding-left: 24px;
  width: 208px;
  display: flex;
  min-height: 56px;
  align-items: center;
  cursor: pointer;
  outline: none;
  text-decoration: none;
  width: 100%;
}

.c2:hover {
  background: #2C3A73;
  color: #FFFFFF;
}

.c1 {
  background: #222C59;
  border-right: 1px solid #111B48;
  min-width: 240px;
  width: 240px;
  overflow: auto;
  height: 100%;
  display: flex;
  flex-direction: column;
}

.c4 {
  display: flex;
  flex-direction: column;
  overflow: auto;
}

<div
  class="c0"
  height="100%"
  style="position: absolute; overflow: hidden;"
>
  <nav
    class="c1"
  >
    <a
      class="c2"
      href="/web"
      pl="4"
      theme="[object Object]"
      width="208px"
    >
      <img
        class="c3"
        src="file_stub"
      />
    </a>
    <div
      class="c4"
    >
      <a
        class="c5"
        href="/web/cluster/one/nodes"
      >
        <span
          class="c6 icon icon-server sc-AxjAm c7"
          color="inherit"
          font-size="16px"
        />
        Servers
      </a>
      <a
        class="c5"
        href="/web/cluster/one/apps"
      >
        <span
          class="c6 icon icon-new-tab sc-AxjAm c7"
          color="inherit"
          font-size="16px"
        />
        Applications
      </a>
      <a
        class="c5"
        href="/web/cluster/one/kubes"
      >
        <span
          class="c6 icon icon-kubernetes sc-AxjAm c7"
          color="inherit"
          font-size="16px"
        />
        Kubernetes
      </a>
      <button
        class="c8 actives"
      >
        <span
          class="c6 icon icon-users2 sc-AxjAm c7"
          color="inherit"
          font-size="16px"
        />
        Team
        <span
          class="c9 icon icon-chevron-down "
          color="inherit"
          style="font-size: 14px;"
        />
      </button>
      <div
        class="c10"
        style="display: block;"
      >
        <a
          class="c11"
          href="/web/users"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-users2 sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Users
        </a>
        <a
          aria-current="page"
          class="c11 active"
          href="/web/roles"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-profile sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Roles
        </a>
        <a
          class="c11"
          href="/web/sso"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-lock sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Auth. Connectors
        </a>
      </div>
      <button
        class="c8"
      >
        <span
          class="c6 icon icon-alarm-ringing sc-AxjAm c7"
          color="inherit"
          font-size="16px"
        />
        Activity
        <span
          class="c9 icon icon-chevron-right "
          color="inherit"
          style="font-size: 14px;"
        />
      </button>
      <div
        class="c10"
        style="display: none;"
      >
        <a
          class="c11"
          href="/web/cluster/one/sessions"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-cli sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Active Sessions
        </a>
        <a
          class="c11"
          href="/web/cluster/one/recordings"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-play-circle sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Session Recordings
        </a>
        <a
          class="c11"
          href="/web/cluster/one/audit"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-list1 sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Audit Log
        </a>
      </div>
      <button
        class="c8"
      >
        <span
          class="c6 icon icon-icons2 sc-AxjAm c7"
          color="inherit"
          font-size="16px"
        />
        Clusters
        <span
          class="c9 icon icon-chevron-right "
          color="inherit"
          style="font-size: 14px;"
        />
      </button>
      <div
        class="c10"
        style="display: none;"
      >
        <a
          class="c11"
          href="/web/clusters"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-equalizer1 sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Manage Clusters
        </a>
        <a
          class="c11"
          href="/web/trust"
        >
          <div
            class="c12 marker"
          />
          <span
            class="c13 icon icon-site-map sc-AxjAm c14"
            color="inherit"
            font-size="2"
          />
          Trust
        </a>
      </div>
    </div>
  </nav>
</div>
`;
